src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1344,9 +1354,48 @@ private static bool IsLessDerivedThanAny<TMember>(int index, TypeSymbol type, Ar
                 {
                     return true;
                 }
-            }
+                else if (currentType.IsExtension)
+                {
+                    if (currentType.ExtendedTypeNoUseSiteDiagnostics is { } extendedType)
+                    {
+                        if (extendedType.Equals(type, TypeCompareKind.ConsiderEverything))
+                        {
+                            return true;
+                        }
 
-            return false;
+                        if (extendedType is TypeParameterSymbol currentTypeParameter)
+                        {
+                            var baseClass = currentTypeParameter.EffectiveBaseClass(ref useSiteInfo);
+                            if (type.Equals(baseClass, TypeCompareKind.ConsiderEverything)
+                                || isLessDerived(type, currentType: baseClass, ref useSiteInfo))
+                            {
+                                return true;
+                            }
+
+                            foreach (var i in currentTypeParameter.AllEffectiveInterfacesWithDefinitionUseSiteDiagnostics(ref useSiteInfo))
```

Comment:
>AllEffectiveInterfacesWithDefinitionUseSiteDiagnostics [](http://example.com/codeflow?start=67&length=54)

We probably have a ```Contains``` helper for ```ImmutableArray```

Reply:
Unless this is essential, I'd rather leave as-is. Thanks